core/application/impl/app_config_impl.cpp

```diff
@@ -0,0 +1,332 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "application/impl/app_config_impl.hpp"
+
+#include <rapidjson/document.h>
+#include <rapidjson/filereadstream.h>
+#include <boost/program_options.hpp>
+#include <iostream>
+
+namespace {
+  struct RegisteredHandlers {
+    using Handler =
+        void (kagome::application::AppConfigurationImpl::*)(rapidjson::Value &);
+    char const *segment_name;
+    Handler handler;
+  };
+
+  template <typename T, typename Func>
+  inline void find_argument(boost::program_options::variables_map &vm,
+                            char const *name,
+                            Func &&f) {
+    assert(nullptr != name);
+    auto it = vm.find(name);
+    if (it != vm.end()) std::forward<Func>(f)(it->second.as<T>());
+  }
+
+  const std::string def_rpc_http_host = "0.0.0.0";
```

Comment:
Please, don't short name wherever then possible, with the exception of generally accepted abbreviations.

Reply:
Another way is use compose struct with commona name, e.g. DefaultValues or DefaultConfig

Reply:
Def is a general accepted abbreviation of default https://imgur.com/pJnyXvt.png

Reply:
As you wish. =) I seem it would be better for readability. =)